Makefile

```diff
@@ -32,22 +32,47 @@ else
 endif
 
 # targets
-build-all: build-darwin build-linux build-windows
-
-build-darwin: generate
-	GOOS=darwin GOARCH=amd64 $(GOBUILD) -v \
-		-ldflags "${LD_FLAGS} -X ${REPO}/pkg/version.OS=darwin -X ${REPO}/pkg/version.Arch=amd64" \
-		-o ./build/darwin/amd64/$(BINARY_NAME)
-
-build-linux: generate
-	GOOS=linux GOARCH=amd64 $(GOBUILD) -v \
-		-ldflags "${LD_FLAGS} -X ${REPO}/pkg/version.OS=linux -X ${REPO}/pkg/version.Arch=amd64" \
-		-o ./build/linux/amd64/$(BINARY_NAME)
-
-build-windows: generate
-	GOOS=windows GOARCH=amd64 $(GOBUILD) -v \
-		-ldflags "${LD_FLAGS} -X ${REPO}/pkg/version.OS=windows -X ${REPO}/pkg/version.Arch=amd64" \
-		-o ./build/windows/amd64/$(BINARY_NAME).exe
+build-all: build-all-darwin build-all-linux build-all-windows
+
+build-all-darwin: build-darwin-amd64 build-darwin-arm64
+
+build-darwin-amd64:
+	GOOS=darwin GOARCH=amd64 make build-binary
+
+build-darwin-arm64:
+	GOOS=darwin GOARCH=arm64 make build-binary
+
+build-all-linux: build-linux-386 build-linux-amd64 build-linux-arm build-linux-arm64
+
+build-linux-386:
+	GOOS=linux GOARCH=386 make build-binary
+
+build-linux-amd64:
+	GOOS=linux GOARCH=amd64 make build-binary
+
+build-linux-arm:
+	GOOS=linux GOARCH=arm make build-binary
+
+build-linux-arm64:
+	GOOS=linux GOARCH=arm64 make build-binary
+
+build-all-windows: build-windows-386 build-windows-amd64
+
+build-windows-386:
+	GOOS=windows GOARCH=386 make build-binary-windows
+
+build-windows-amd64:
+	GOOS=windows GOARCH=amd64 make build-binary-windows
+
+build-binary:
+	CGO_ENABLED="0" GOOS=$(GOOS) GOARCH=$(GOARCH) $(GOBUILD) -v \
```

Comment:
Can we re-enable CGO on Linux platforms? That would improve detecting user's `$HOME` directory in cases when the env var isn't set. What was the original reason for disabling CGO again?

Reply:
Asking because CGO would fix https://github.com/wakatime/vscode-wakatime/pull/238

Reply:
https://github.com/wakatime/wakatime-cli/pull/547